src/features/textEditor/components/TipTapEditor.tsx

```diff
@@ -81,8 +84,11 @@ export function TipTapEditor({ editorContent, editorId, isActive, saveFileInMemo
     const mdSerializer = new MarkdownSerializer(editor, references);
     const { id: filePath } = parseEditorId(editorId);
 
-    void saveAsMarkdown(mdSerializer, filePath);
-  }, [editor, editorId, isActive, references]);
+    void saveAsMarkdown(mdSerializer, filePath).then(() => {
+      // We need to make sure the files were saved before refreshing the file tree.
+      void refreshFileTree();
```

Comment:
This was missing.

Reply:
Nice thank you for that!
You can simplify it like this:
```typescript
void saveAsMarkdown(mdSerializer, filePath)
      // We need to make sure the files were saved before refreshing the file tree.
      .then(refreshFileTree);
```